src/structures/GuildEmoji.js

```diff
@@ -30,7 +30,7 @@ class GuildEmoji extends BaseGuildEmoji {
      * @type {Snowflake[]}
      * @private
      */
-    Object.defineProperty(this, '_roles', { value: [], writable: true });
```

Comment:
Oh, this happens because _patch is called before defining the field? Is that the case? If _patch isn't called at all, thats the bug that should be fixed instead 👀

Reply:
Ah yeah, always calling `._patch()` will also correct the author data if not known beforehand - thank you! Should I leave `data.roles` as the default value here, as it seems to be always present (and an empty array if no roles are provided)?

Reply:
I think you can leave it as an empty array by default and call _patch, do test this tho

Reply:
Empty array or default array whilst calling `._patch()` works fine in the event that:
1. Emoji is simply not cached but then fetched from the manager (role and author data is populated as expected)
2. Emoji is received from the gateway (author data not present and is `null` as expected, roles populated as expected)
3. Emoji is cached without author data, but fetching with `{ force: true }` or fetching the author populates the `.author` property

I'll amend `data.roles` back to `[]` and call `this._patch()`!